src/cfg/to_bril.rs

```diff
@@ -0,0 +1,129 @@
+        };
+        for func in &self.functions {
+            bril.functions.push(func.to_bril());
+        }
+        bril
+    }
+}
+
+impl SimpleCfgFunction {
+    /// Converts the cfg function into a bril program.
+    pub fn to_bril(&self) -> Function {
+        // Make an empty function
+        let mut func = Function {
+            name: self.name.clone(),
+            args: self.args.clone(),
+            instrs: vec![],
+            pos: None,
+            return_type: self.return_ty.clone(),
+        };
+
+        // start with the entry block
+        self.node_to_bril(self.entry, &mut func);
+
+        // The order of this traversal does not matter, just need to loop over the blocks
+        DfsPostOrder::new(&self.graph, self.entry)
+            .iter(&self.graph)
+            .for_each(|node| {
+                // don't do the exit or entry
+                if node != self.entry && node != self.exit {
+                    // Add a label for the block
+                    func.instrs.push(Code::Label {
+                        label: format!("{}", self.graph[node].name),
+                        pos: None,
+                    });
+                    // rest of the block
+                    self.node_to_bril(node, &mut func);
+                }
+            });
+
+        // now do the exit at the end
+        func.instrs.push(Code::Label {
+            label: format!("{}", self.graph[self.exit].name),
+            pos: None,
+        });
+        self.node_to_bril(self.exit, &mut func);
+
+        func
+    }
+
+    // Converts a node to bril, including jumps at the end
+    // Doesn't add the label for the node
+    fn node_to_bril(&self, node: NodeIndex, func: &mut Function) {
+        let block = &self.graph[node];
+        func.instrs.extend(block.to_bril());
+
+        // now add the jump to another block
+        match self.get_branch(node) {
+            SimpleBranch::NoBranch => {}
+            SimpleBranch::Jmp(to) => {
+                func.instrs.push(Code::Instruction(Instruction::Effect {
+                    op: EffectOps::Jump,
+                    args: vec![],
+                    labels: vec![format!("{}", to)],
+                    funcs: vec![],
+                    pos: None,
+                }));
+            }
+            SimpleBranch::If {
+                arg,
+                then_branch,
+                else_branch,
+            } => {
+                func.instrs.push(Code::Instruction(Instruction::Effect {
+                    op: EffectOps::Branch,
+                    args: vec![arg.to_string()],
+                    labels: vec![format!("{}", then_branch), format!("{}", else_branch)],
+                    funcs: vec![],
+                    pos: None,
+                }));
+            }
+        }
+    }
+}
+
+impl BasicBlock {
+    // Converts this block to bril, not including jumps to other block
+    // or edges.
+    pub fn to_bril(&self) -> Vec<Code> {
+        let mut res = vec![];
```

Comment:
nit, optional: `let mut res: Vec<Code> = self.instrs.iter().cloned().map(Code::Instruction).collect();`

Allows the vector to be pre-sized correctly.

Reply:
Didn't do this one